internal.nix

```diff
@@ -274,6 +290,11 @@ rec {
             '';
           executable = true;
         };
+
+        sourceHashFunc = spec:
```

Comment:
This deserves a docstring and perhaps moving out of the `node_modules`  function. We could have a `mkSourceHashFunc :: Map -> Fn` which we could even properly unit test.